### PR TITLE
revise atan op

### PR DESCRIPTION
**Pull request by songbell**
### Details:
 - Add detailed description to Atan Spec
 - Change to use RTTI decl/define for Atan OP
 - Add type prop , visitor (attributes) tests for Atan OP.

### Tickets:
 - 37421

### CODE REVIEW

File: ngraph/test/visitors/op/atan.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#include "util/visitor.hpp"
+
+using namespace std;
+using namespace ngraph;
+using ngraph::test::NodeBuilder;
+
+TEST(attributes, atan_op)
+{
+    NodeBuilder::get_ops().register_factory<op::Atan>();
```
Comment (ilyachur) [inline ngraph/test/visitors/op/atan.cpp:19]: ould you please rebase your PR to the latest master. And please use parametrized visitor API test which was introduced in this PR: #6181

Comment (songbell) [reply ngraph/test/visitors/op/atan.cpp]: fixed, changed to use parametrized visitor API test

**Comment by ilyachur** [comment]
Please resolve merge conflicts.

File: ngraph/test/visitors/op/atan.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "unary_ops.hpp"
+
+using Types = ::testing::Types<UnaryOperatorType<ngraph::op::v0::Atan, element::f32>>;
+
+INSTANTIATE_TYPED_TEST_CASE_P(visitor_without_attribute,
```
Comment (ilyachur) [inline ngraph/test/visitors/op/atan.cpp:9]: Please rebase your PR to the latest master and use `INSTANTIATE_TYPED_TEST_SUITE_P` macro

Comment (songbell) [reply ngraph/test/visitors/op/atan.cpp]: done rebase, use new macro for the test

File: ngraph/test/op_eval/atan.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#include "ngraph/op/atan.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/validation_util.hpp"
+#include "runtime/backend.hpp"
+#include "util/test_tools.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(op_eval, atan)
```
Comment (ggalieroc) [inline ngraph/test/op_eval/atan.cpp:19]: Please use backend tests instead. Thread here: https://github.com/openvinotoolkit/openvino/pull/6361#discussion_r660482249

**Review by ggalieroc — CHANGES_REQUESTED** [review]

File: docs/ops/arithmetic/Atan_1.md [hunk 4]
```diff
@@ -2,10 +2,16 @@
...
 
+**Detailed description**:  Operation takes one input tensor and performs the element-wise inverse tangent function on a given input tensor, based on the following mathematical formula:
+
+\f[
+a_{i} = atan(a_{i})
+\f]
+
 **Attributes**:
 
     No attributes available.
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Atan_1.md:17]:
```suggestion
**Attributes**: *Atan* operation has no attributes.
```

File: docs/ops/arithmetic/Atan_1.md [hunk 5]
```diff
@@ -22,12 +28,6 @@
 
 * *T*: any numeric type.
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Atan_1.md:29]:
```suggestion
* *T*: any supported numeric type.
```

File: docs/ops/arithmetic/Atan_1.md [hunk 6]
```diff
@@ -22,12 +28,6 @@
...
 
-*atan* does the following with the input tensor *a*:
-
-\f[
-a_{i} = atan(a_{i})
-\f]
-
 **Examples**
 
 *Example 1*
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Atan_1.md:33]: When there is only one example, what we are doing is cleaning up \*Example 1\* and leave just \*\*Examples\*\*

**Review by ggalieroc — COMMENTED** [review]
Please correct Inputs/Output sections as follows:

\*\*Inputs\*\*
\* \*\*1**: A tensor of type \*T\* and arbitrary shape. \*\*Required.\*\*
\*\*Outputs\*\*
\* \*\*1\*\*: The result of element-wise \*Atan\* operation applied to the input tensor. A tensor of type \*T\* and same shape as the input tensor.

Summarizing:
- Types have to be surrounded by \*
- Always specify type and shape of inputs/outputs

File: ngraph/core/src/op/atan.cpp [hunk 7]
```diff
@@ -23,7 +23,7 @@
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::Atan::type_info;
+NGRAPH_RTTI_DEFINITION(op::v0::Atan, "Atan", 0);
```
Comment (ggalieroc) [inline ngraph/core/src/op/atan.cpp:26]:
```suggestion
NGRAPH_RTTI_DEFINITION(op::v0::Atan, "Atan", 0, util::UnaryElementwiseArithmetic);
```

**Comment by ggalieroc** [comment]
Additionally, this operation as spec indicates, should have reference implementation specific for integer types. Please check #6429 for implementation details

File: docs/ops/arithmetic/Atan_1.md (hunk 6 above)
Comment (songbell) [reply docs/ops/arithmetic/Atan_1.md]: updated in doc.

File: ngraph/test/op_eval/atan.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (songbell) [reply ngraph/test/op_eval/atan.cpp]: per #6361, drop the op_eval test, been covered in backend test

Comment (songbell) [reply ngraph/test/op_eval/atan.cpp]: per #6361, drop the op_eval test, been covered in backend test

[the remark this answers is not in the document]
Comment (ilyachur) [reply ngraph/test/op_eval/atan.cpp]: I mean, could you verify that we have SLT for this operation?

[the remark this answers is not in the document]
Comment (songbell) [reply ngraph/test/op_eval/atan.cpp]: yes, the single layer test is in inference-engine/tests/functional/inference_engine/serialization/single_layer/activation.cpp

**Review by ilyachur — COMMENTED** [review]
Looks like `ngraph/test/type_prop/atan.cpp` is empty file

File: docs/ops/arithmetic/Atan_1.md [hunk 8]
```diff
@@ -6,32 +6,28 @@
...
+\f[
+a_{i} = atan(a_{i})
+\f]
+
+**Attributes**: *Atan* operation has no attributes.
 
 **Inputs**
 
-* **1**: An tensor of type *T*. **Required.**
+* **1**: An tensor of type *T* and arbitrary shape. **Required.**
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Atan_1.md:19]: typo: An tensor of ---> A tensor of ...

**Review by ggalieroc — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**